### DISCUSSION TITLE
Site won't deploy; "No such file or directory - jupyter"

### DISCUSSION THREAD

**Discussion by lvanveen** [alshedivat/al-folio#2495] [Q&A] [answered]
I downloaded the latest version of the alshedivat/al-folio template and followed the instructions in INSTALL.md. In addition, I had to switch the Build and Deployment source (see https://github.com/alshedivat/al-folio/issues/1398#issuecomment-1558477269) and add a Repository Secret to insert my Personal Access Token (see https://github.com/alshedivat/al-folio/issues/2122#issuecomment-1910312050). After all that, the deployment fails with "converter.rb:58:in `spawn': No such file or directory - jupyter (Errno::ENOENT)" and the error message in the title of this post. I have seen dome discussion around this (see https://github.com/alshedivat/al-folio/issues/1545#issue-1795123175) but no solution. I am quite stuck. This happens when running the deployment on GitHub, not locally.
If you have an idea how to fix this, please explain it in detail. I am new to Jekyll so I need some hand-holding.

**Comment by george-gca** [comment]
Have you followed the recommended approach first? You should first make it work "as is" in your repository, only then make modifications. This way it is easier to keep track of which modifications caused error.

**Reply by lvanveen** [reply]
Yes, that is exactly what I have done. The only modifications I made are listed in the question. Otherwise, the site repository is as I found it.

**Reply by george-gca** [reply, marked as answer]
You should use `Deploy from a branch`, not what is stated in that comment. Note that it takes some time for the GitHub action that does the build and create `gh-pages` branch to run, which you can check in the `Actions` tab of your repo.

Also we don't use `.github/workflows/jekyll.yml`, since we have our own build action in [.github/workflows/deploy.yml](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/deploy.yml). That action installs `jupyter` before building anything. Delete your `.github/workflows/jekyll.yml` file and try again, changing to `Deploy from a branch`.

**Reply by lvanveen** [reply]
OK, I started afresh with a new copy of the template. This time, the first deployment has a couple of red crosses, but a new branch "gh" is created. When I change to deployment from that branch, the red crosses disappear but a new one appears flagging the Lighthouse Badger issue. Adding the PAT as a LIGHTHOUSE_BADGER_TOKEN solves that. I now see the template appear under my URL and can start editing. The only error that keeps coming up now is
___
Run npx prettier . --check
  npx prettier . --check
  shell: /usr/bin/bash -e {0}
Checking formatting...
[warn] _config.yml
[warn] Code style issues found in the above file. Run Prettier to fix.
Error: Process completed with exit code 1.
___
but it does not seem to stop the process of publishing.

**Reply by george-gca** [reply]
For this see our [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md#my-code-runs-fine-locally-but-when-i-create-a-commit-and-submit-it-it-fails-with-prettier-code-formatter-workflow-run-failed-for-master-branch-how-do-i-fix-that). But yes, prettier errors do not stop the publishing process.
